app/src/main/res/layout/fragment_trip_detail.xml

```diff
@@ -87,24 +89,105 @@
 				android:singleLine="true"
 				android:textColor="?android:textColorPrimary"
 				app:layout_constraintBottom_toBottomOf="parent"
-				app:layout_constraintEnd_toStartOf="@+id/duration"
+				app:layout_constraintEnd_toStartOf="@+id/durationPrice"
 				app:layout_constraintStart_toEndOf="@+id/toTime"
 				app:layout_constraintTop_toBottomOf="@+id/from"
 				tools:text="To Station"/>
 
+			<LinearLayout
+				android:id="@+id/durationPrice"
+				android:layout_width="wrap_content"
+				android:layout_height="match_parent"
+				android:orientation="vertical"
+				android:layout_marginTop="8dp"
+				android:layout_marginEnd="8dp"
+				android:layout_marginRight="8dp"
+				android:layout_marginBottom="8dp"
+				android:gravity="center_vertical"
+				app:layout_constraintEnd_toEndOf="parent"
+				app:layout_constraintTop_toTopOf="parent"
+				app:layout_constraintBottom_toBottomOf="parent">
```

Comment:
Would be nice to not introduce another hierarchy here, but use the existing ConstraintLayout for layouting.

Reply:
I'm not sure if that is really feasible because the price might not be shown if the information is not available. At least in my tries I didn't manage to layout it nicely in both cases with and without price.

Reply:
What issue specifically did you have? Note there's other views that get hidden and shown even dynamically. The ConstraintLayout should give you everything you need for avoiding nested layouts here.

Reply:
Thanks for insisting, after some searching in the android docs I found ConstraintLayout Barriers to be quite useful for these kind of problem. Should be fixed with the last commit.